### PR TITLE
[multistage]: better error messages.

### PR DESCRIPTION
**Pull request by abhioncbr**
- handling the `json` data type in a multistage query engine. as per the [comment](https://github.com/apache/pinot/issues/10205#issuecomment-1543278577) 
- returning the meaningful error msg for unsupported type. 
- returning the meaningful error message of SQL parsing errors.

As per the issue https://github.com/apache/pinot/issues/10205

cc @walterddr

### CODE REVIEW

**Comment by kishoreg** [comment]
LGTM.. lets add a test case

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java [hunk 1]
```diff
@@ -108,6 +108,9 @@ public String handlePostSql(String requestJsonStr, @Context HttpHeaders httpHead
       }
       LOGGER.debug("Trace: {}, Running query: {}", traceEnabled, sqlQuery);
       return executeSqlQuery(httpHeaders, sqlQuery, traceEnabled, queryOptions, "/sql");
+    } catch (ProcessingException pe) {
+      LOGGER.error("Caught exception while processing get request {}", pe.getMessage());
```
Comment (walterddr) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java:112]: this is a POST request not a get request

File: pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java [hunk 2]
```diff
@@ -81,14 +81,14 @@ private RelDataType toRelDataType(FieldSpec fieldSpec) {
         return fieldSpec.isSingleValueField() ? createSqlType(SqlTypeName.DECIMAL)
             : createArrayType(createSqlType(SqlTypeName.DECIMAL), -1);
       case JSON:
-        // TODO: support JSON, JSON should be supported using a special RelDataType as it is not a simple String,
-        // nor can it be easily parsed as a STRUCT.
+        return createSqlType(SqlTypeName.VARCHAR);
```
Comment (walterddr) [inline pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java:84]: let's separate the support of the JSON type into a separate PR. and for this PR let's only contribute the proper error message with unsupported datatype propagation.

**Review by walterddr — COMMENTED** [review]
lgtm overall. please see my comments. let's separate supporting JSON and error propagation into 2 different PRs, as they address 2 different problems.

**Review by walterddr — COMMENTED** [review]
lgtm overall. please see my comments. let's separate supporting JSON and error propagation into 2 different PRs, as they address 2 different problems.

File: pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java (hunk 2 above)
Comment (abhioncbr) [reply pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java:84]: Make sense.

**Review by walterddr — APPROVED** [review]
lgtm

**Pull request merged**